### ISSUE TITLE
Javascript Tests Won't run in browser or PhantomJS

### ISSUE THREAD

**Issue by vipero07** [ASP-NET-Core-Boilerplate/Templates#179]
[Related issue opened on another project by Rehan Saeed](https://github.com/mrhooray/gulp-mocha-phantomjs/issues/47)

I figured out why this fails, and the problems are two fold.

1. Sinon doesn't seem to have a framework compatible with browsers (it throws an error in the console, specifically "require is not defined" because it is using the nodejs require methods). Commenting out Sinon from the mocha.html helps, it should probably be imported in the site-test.js anyhow (which should also probably be a .ts file).
2. The calculator class that is exported isn't inside a namespace (or module) in the console this causes a similar problem to sinon (though i think it is "export is undefined").  In the mocha.html page the globals set are "framework" and in site-test.js it uses new `framework.Calculator()`. A simple fix is to wrap the entire `export class Calculator {}` in `namespace framework { }` i.e. `namespace framework { export class Calculator {...}}`

Once those changes are made you can both open mocha.html in a browser and run the gulp test task successfully.

﻿**Which project template is the issue about?**

ASP.NET Core 1.1.2 Boilerplate (VS 2017 Only)

**Which version of the project template is the issue about?**

Boilerplate ASPNetCore 2.2.3

**Which version of Visual Studio or the dotnet SDK were you using?**

2017 15.3

**Comment by RehanSaeed** [comment]
The `namespace` keyword is a TypeScript only construct that I would not recommend using.

https://stackoverflow.com/questions/38582352/module-vs-namespace-import-vs-require-typescript

We should be using the `export` and `import` keywords instead.

**Comment by vipero07** [comment]
Agreed, but using export and import would either need commonjs or some similar library loaded, or to change the typescript module type from commonjs to es6. Making the modules es6 has other issues though particularly CORS blocks js modules from being loaded into the browser when loaded from the filesystem instead of a server, so either there would need to be some intermediary type thing like webpack or a server would need to be brought up to run the tests.

The .net core preview 2 react/redux template seems to rely on webpack and [awesome-typescript-loader](https://www.npmjs.com/package/awesome-typescript-loader). In that way the TS files are not being generated on save and they are able to import and export without name spacing.

**Comment by RehanSaeed** [comment]
I've archived the MVC boilerplate project due to the size and complexity of moving it to ASP.NET Core 2.0 with the latest and greatest web front end stuff like Bootstrap 4, Webpack, Workbox etc. I also had to learn all that stuff so needed time for that. I'm focusing my efforts on the API Boilerplate project and will bring back the MVC project better than ever before once I have the courage to attempt it.
